using_restricted_array.rb

```diff
@@ -7,41 +7,104 @@
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
 def length(array)
-  puts "NOT IMPLEMENTED"
+  i = 0
+  value = array[0]
+  until value == nil
+    value = array[i]
+    i += 1
+  end
+  return (i - 1)
```

Comment:
You should return i and not i-1. Consider a scenario where there are 3 elements in the array. When array[i] is nil, i will be 3 which is the length of the array. Returning 2 will be inaccurate.